### PR TITLE
dev/sg: 'sg ci' build targeting improvements

### PR DESCRIPTION
**Pull request by bobheadxi**
I want to make `sg ci build main-dry-run` available for contractors, and also was inspired to give this another pass by @burmudar 's work on `sg ci` annotations, so did some additional cleanup here alongside the pipeline configuration to improve "build targeting/selecting":

- `-pipeline` can now be configured (also via env var), for e.g. `sourcegraph-contractors`
- using `-branch`, `-build` for selecting target builds works more consistently now
- fix issue with `-wait` where it might start selecting the wrong build while polling

## Test plan

`go run ./dev/sg ci status -p sourcegraph-contractors -branch contractors/SG-36384`

and also test out the other `sg ci` commands

### CODE REVIEW

File: dev/sg/sg_ci.go [hunk 1]
```diff
@@ -32,37 +35,88 @@ const (
...
+		Aliases: []string{"b"},
+		Usage:   "Branch `name` of build to target (defaults to current branch)",
 	}
-
-	ciBuild     string
 	ciBuildFlag = cli.StringFlag{
-		Name:        "build",
-		Usage:       "Override branch detection with a specific build `number`",
-		Destination: &ciBuild,
+		Name:    "build",
```
Comment (jhchabran) [inline dev/sg/sg_ci.go:44]: FWIW, this is referenced as build number in Buildkite docs, so perhaps `-buildnum` is a better name for this flag.

**Review by jhchabran — APPROVED** [review]
Ah nice! Did the contractors asked for this? Just wondering :)

File: dev/sg/sg_ci.go (hunk 1 above)
Comment (bobheadxi) [reply dev/sg/sg_ci.go:44]: That's a bit of a clunky name - right now, the help text indicates number, which I think should be sufficient:

image

Comment (jhchabran) [reply dev/sg/sg_ci.go:44]: np np, that was merely a nit suggestion :)

**Pull request merged**